src/main/java/com/github/dbmdz/solrocr/model/SourcePointer.java

```diff
@@ -129,4 +105,63 @@ private static SourcePointer.Region parseRegion(String r) {
   public SourcePointer(List<FileSource> sources) {
     this.sources = sources;
   }
+
+  /**
+   * Create meanigful human-readable representation of {@link SourcePointer} from it's attached
+   * files
+   */
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder();
+    for (FileSource source : this.sources) {
+      if (sb.length() > 0) {
+        sb.append("+");
+      }
+      sb.append(source.path);
+      if (source.isAscii) {
+        sb.append("{ascii}");
+      }
+      if (!source.regions.isEmpty()) {
+        sb.append("[");
+        for (Region region : source.regions) {
+          if (sb.charAt(sb.length() - 1) != '[') {
+            sb.append(",");
+          }
+          sb.append(region.start);
+          sb.append(":");
+          sb.append(region.end);
+        }
+        sb.append("]");
+      }
+    }
+    return sb.toString();
+  }
+}
+
+/** Explicite Mapper Implementation */
+class ToFileSource implements Function<String, SourcePointer.FileSource> {
```

Comment:
This would be better as a static factory method `parse` on `SourcePointer.FileSource` that you can then reference in the `.map` in L86:

```java
List<FileSource> = Arrays.stream(sourceTokens).map(FileSource::parse).collect(Collectors.toList());
```

Reply:
This seems rather a philosophical question, but if you want it this way, it's fine.
One could even ask why no `SourcePointerFactory` since under the hood these are allocated from `List<FileSource>`  :slightly_smiling_face: